internal/action/find.go

```diff
@@ -141,13 +144,21 @@ func (s *Action) findSelection(ctx context.Context, c *cli.Context, choices []st
 	}
 }
 
-func filter(l []string, needle string) []string {
+func filter(l []string, needle string, regex bool) ([]string, error) {
 	choices := make([]string, 0, 10)
 	for _, value := range l {
-		if strings.Contains(strings.ToLower(value), needle) {
+		if regex {
+			matched, err := regexp.MatchString(needle, value)
+			if err != nil {
+				return nil, err
+			}
+			if matched {
+				choices = append(choices, value)
+			}
+		} else if strings.Contains(strings.ToLower(value), strings.ToLower(needle)) {
```

Comment:
`else if` is highly unusual in Go and at least in a loop rarely needed. I would suggest to handle the regexp case and at the end of that block add a `continue` so the non-regexp part is skipped in that case.

May if would make sense so swap the order of both cases, but I'm sure about that.